scripts/deconvolution_funs.R

```diff
@@ -223,31 +223,38 @@ get_protein_mut <- function(vepfile) {
 }
 
 
-createSigMatrix <- function(mutations.vector, mutation_sheet) {
+createSigMatrix <- function(mutations_vector, mutation_sheet_file) {
   #' for making the signature matrix based on the signature mutations found in
   #' the sample (given as input as a vector)for it self
   #' returns simple signature matrix as data.frame without frequency values
-  
+
   # read in provided mutation sheet
-  mutations.df <- read.csv(mutation_sheet)
-  if ("source" %in% colnames(mutations.df)){
-      mutations.df <- mutations.df[,-(which(names(mutations.df) %in% "source"))]
-  }
-  # create an empty data frame add a column for the Wildtype
-  # "Others" means that the particular mutation is not found and the mutation site could be mutated otherwise or not at all
-  
-  msig <- setNames( data.frame( matrix( ncol = ncol(mutations.df)+1, nrow = 0 )), c("Others", colnames(mutations.df)))
-  msig <- bind_rows(tibble(muts=mutations.vector), msig)
+  mutations_df <- read.csv(mutation_sheet_file) %>%
+
+    # remove source col if there. TODO What problems would this col cause?
+    dplyr::select(-matches("source"))
+
   # making a matrix with the signature mutations found in the sample
-  # make binary matrix matching the mutations to the mutation-list per variant to see how many characterising mutations
-  # where found by variant
-  
-  for ( variant in names(mutations.df) ){
-    msig[[ variant ]] <- msig$muts %in% mutations.df[[ variant ]]
-  }
-  msig[is.na(msig)] <- 0
-  
-  return( msig[,-match('muts', names(msig))]*1 ) # use the *1 to turn true/false to 0/1
+  # make binary matrix matching the mutations to the mutation-list per variant
+  # to see how many characterising mutations where found by variant
+
+  sig_mat <- lapply(colnames(mutations_df), function(variant) {
+    return(as.numeric(mutations_vector %in% mutations_df[[variant]]))
+  }) %>%
+
+    dplyr::bind_cols() %>%
+
+    magrittr::set_names(names(mutations_df)) %>%
+
+    # add "Others" col of all "0"s to indicate possible other variants not
+    # possessing any of the mutations
+    # TODO ensure this is how this column is supposed to work
+    mutate(Others = rep(0, length(mutations_vector))) %>%
```

Comment:
Yes and that is in line with how simulateOthers worked, because then in the logical inverse matrix it becomes all 1, simulating that variant having the NULL mutations at the loci where other variants have actual mutations.